### ISSUE TITLE
Problems opening OpenGL captures if 'Ref All Resources' is enabled

### ISSUE THREAD

**Issue by andre-ferreira** [baldurk/renderdoc#1068]
## Description

I cant start my app in the version 1.1 of renderdoc. Version 1.0 still works fine

## Repro steps

- Open renderdoc
- Set application path
- Launch application
- Application closes immediately

This is the log:
```
RDOC 027368: [11:03:57]             core.cpp( 300) - Log     - RenderDoc v1.1 64-bit Release (c7d5aa863e5f13430b2230e5048d253d342d1a18) loaded in replay application
RDOC 022400: [11:03:57]             core.cpp( 300) - Log     - RenderDoc v1.1 64-bit Release (c7d5aa863e5f13430b2230e5048d253d342d1a18) loaded in replay application
QTRD 027368: [11:03:57]       qrenderdoc.cpp(  88) - Log     - QRenderDoc initialising.
RDOC 002676: [11:03:57]             core.cpp( 300) - Log     - RenderDoc v1.1 64-bit Release (c7d5aa863e5f13430b2230e5048d253d342d1a18) loaded in replay application
QTRD 027368: [11:04:04]              unknown(   0) - Debug   - QWindowsNativeFileDialogBase::onSelectionChange () 0
QTRD 027368: [11:04:04]              unknown(   0) - Debug   - QWindowsNativeFileDialogBase::onSelectionChange () 0
QTRD 027368: [11:04:05]              unknown(   0) - Debug   - QWindowsNativeFileDialogBase::onSelectionChange (QUrl("file:///C:/Users/afferreira/source/repos/OpenCars/build/mOpenGLdisplay/app/Release/display.exe")) 1
RDOC 027368: [11:04:07]    win32_process.cpp( 490) - Log     - Running process C:\Users\afferreira\source\repos\OpenCars\build\mOpenGLdisplay\app\Release\display.exe
RDOC 027368: [11:04:07]    win32_process.cpp( 574) - Log     - Injecting renderdoc into process 22068
RDOC 022068: [11:04:07]             core.cpp( 300) - Log     - RenderDoc v1.1 64-bit Release (c7d5aa863e5f13430b2230e5048d253d342d1a18) capturing application
RDOC 022068: [11:04:09]   win32_libentry.cpp(  65) - Log     - Loading into C:\Users\afferreira\source\repos\OpenCars\build\mOpenGLdisplay\app\Release\display.exe
RDOC 022068: [11:04:09]  sys_win32_hooks.cpp(  82) - Log     - Registering Win32 system hooks
RDOC 022068: [11:04:09]      d3d11_hooks.cpp( 125) - Log     - Registering D3D11 hooks
RDOC 022068: [11:04:09]      d3d12_hooks.cpp(  91) - Log     - Registering D3D12 hooks
RDOC 022068: [11:04:09]       d3d8_hooks.cpp(  36) - Log     - Registering D3D8 hooks
RDOC 022068: [11:04:09]       d3d9_hooks.cpp(  45) - Log     - Registering D3D9 hooks
RDOC 022068: [11:04:09]       dxgi_hooks.cpp(  77) - Log     - Registering DXGI hooks
RDOC 022068: [11:04:09]        egl_hooks.cpp( 580) - Log     - Registering EGL hooks
RDOC 022068: [11:04:09]         gl_hooks.cpp( 168) - Log     - Registering OpenGL hooks
RDOC 022068: [11:04:09]        wgl_hooks.cpp( 537) - Log     - Registering WGL hooks
RDOC 022068: [11:04:09]         vk_layer.cpp(  53) - Log     - Registering Vulkan hooks
RDOC 027368: [11:04:12]   target_control.cpp( 512) - Log     - Got remote handshake: display [22068]
RDOC 027368: [11:04:12]    win32_network.cpp( 222) - Warning - recv: WSAECONNRESET: An existing connection was forcibly closed by the remote host.
```

## Environment

* RenderDoc build: 1.1 stable.
* Operating System: Windows
* API: OpenGL

**Comment by dos1** [comment]
Dunno if it's related, but I just wanted to use RenderDoc 1.1 on GNU/Linux and got similar results.

gdb shows a crash in `#1  0x00007ffff71f465b in glXGetProcAddress () from /usr/bin/../lib/librenderdoc.so`. I'll try to rebuild RenderDoc and retry with debug symbols later.

**Comment by baldurk** [comment]
@andre-ferreira can you share your program either here or privately with me? The log you've pasted doesn't list any errors so I can't tell what's actually wrong. All I can see is the program exiting, and assuming the crash reporter isn't kicking in that could be anything.

If you attach to your program after starting it (you can use the 'debugger delay' option to give you time to attach before it runs) can you get a callstack to find out what is actually crashing or failing?

---

@dos1 it's very unlikely to be related as the GLX code only runs on linux. Please check with latest code, there were some fixes since v1.1 to the GLX code.

**Comment by andre-ferreira** [comment]
@baldurk  I'm not sure i can provide you with the program because i'm still figuring out some stuff (ocean fft) and i dont want you to see this abomination yet 🤣.

But in the meantime i tried to inject to my program (using a cin before the opengl library init) and it connects but as soon as i give the program input to proceed it immediately crashes with a crash reporter dialog. I'm attaching the zip file here.

Cheers!
[renderdoc_report.zip](https://github.com/baldurk/renderdoc/files/2273706/renderdoc_report.zip)

**Comment by baldurk** [comment]
@andre-ferreira for what it's worth, in the kindest possible way what your program is doing or how it's written doesn't matter to me. I've seen it all, I only care about fixing the bug :smile:.

Perhaps the crash reporter wasn't kicking in because the crash happened too early in process startup, I'm not sure how well it deals with that edge case. Looking at the minidump in that report, is it possible that you're calling glGetIntegerv before you've created a GL context? The crash is inside glGetIntegerv with no driver/context registered, and I don't see a context creation message in the log itself.

**Comment by andre-ferreira** [comment]
@baldurk hahahahaha yes i realize that but i would like to keep that to a last resource type of thing 😃 

I found the problem. As i was searching for that command in my solution (which i didnt find) i found this suspicious command "glDisable(GL_MULTISAMPLE);" before the glew library was initialized. Now i can start the program with no issues on 1.1.

But it makes me wonder how did it even run in the first place 🤔 🤣.

Many thanks for your great tool which i will definitely continue to use! And sorry for taking your time

Cheers,
André

**Comment by andre-ferreira** [comment]
@baldurk Hum, theres something else wrong now too. In version 1.0 i could see everything divided by my debug tags. Now it shows everything divided by colour and depth passes and i cant see anything in the "pipeline state" tab and the timings are also wrong. The log now shows a mountain of errors (attached)

Thanks!
[RenderDoc_2018.08.09_11.39.17.log](https://github.com/baldurk/renderdoc/files/2274468/RenderDoc_2018.08.09_11.39.17.log)

**Comment by baldurk** [comment]
If you can share the capture I might be able to figure out some of what's wrong there, but judging from those errors some of it went wrong at capture time so I think I'll need the application to properly investigate.

If you don't want to share it I'll need your help to debug it another way by building & running a debug branch of renderdoc - if that's the route you want to take send me an [email](mailto:baldurk@baldurk.org) and we can co-ordinate.

**Comment by andre-ferreira** [comment]
@baldurk  I have sent you an email with the program attached, if you run with any issues please let me know, thanks!

**Comment by baldurk** [comment]
Resolved over email, this was due to enabling 'Ref All Resources' and was fixed recently in 98e4a5b613f71ed7985b9b8e8b6f8ae2e16fcd44.
